origin-growth/src/apollo/schema.js

```diff
@@ -0,0 +1,96 @@
+const { gql } = require('apollo-server-express')
+
+const typeDefs = gql`
+  scalar JSON
+
+  ######################
+  #
+  # Query output schema.
+  #
+  ######################
+
+  enum CampaignStatus {
+    pending                   #not yet started
+    active
+    capReached
+    completed
+  }
+
+  enum ActionStatus {
+    inactive
+    active
+    exhausted
+    completed
+  }
+
+  enum ActionType {
+    basicProfile
+    attestations
+    referral
+  }
+
+  enum InviteStatus {
+    pending
+    successful
+  }
+
+  type Price {
+    currency: String!
+    amount: String!
+  }
+
+  type Invite {
+    status: InviteStatus!
+    walletAddress: ID!
+    contactName: String
+    reward: Price
+  }
+
+  interface BaseAction {
+    type: ActionType!
+    status: ActionStatus!
+    rewardEarned: Price
+    rewardPending: Price      # applicable for referral action
+    rewardInfo: Price!        # information about reward
```

Comment:
Is that the amount for each individual reward ? nit: maybe call this just reward ?

Reply:
The thought process is that each action shows the amount of reward that can be earned and the amount that is earned. These two can be of different values e.g. referral shows 10 OGN (per referral) and the amount earned can be [higher than that](https://app.zeplin.io/project/59fa2311bac7acbc8d953da9/screen/5c34963bb7b2dabdd5509888).

On another thought I will move `rewardPending` out of the `BaseAction` interface since only referral has a pending reward ATM.

I was also in dilemma between calling it `rewardInfo` or `reward`. OK, lets go with `reward` for now.